amethyst_assets/src/asset.rs

```diff
@@ -51,6 +49,8 @@ where
     fn is_shared(&self) -> bool;
 }
 
+pub type AssetFuture<A> = Box<Future<Item=A, Error=BoxedErr>>;
```

Comment:
In order to `clone` those you may add `trait FutureClone: Future + Clone`
And then wrap in in `struct AssetFuture(Box<FutureClone<Item=A, Error=BoxedErr>>)`